mlir/test/Target/Cpp/declare_func.mlir

```diff
@@ -0,0 +1,16 @@
+// RUN: mlir-translate -mlir-to-cpp %s | FileCheck %s
+
+// CHECK: int32_t bar(int32_t [[V1:[^ ]*]]);
+emitc.declare_func @bar
+// CHECK: int32_t bar(int32_t [[V1:[^ ]*]]) {
+emitc.func @bar(%arg0: i32) -> i32 {
```

Comment:
Can you change this to call @foo below, like in the op description. This shows the usefulness/need of the op better.

Reply:
So far most of our test "just" test the "core" functionality, whereas transporting the usefulness is left to the description. I agree that this would make a more meaningful example and we should provide examples somewhere. However, till now I was not sure if should use the tests for this. With this thoughts in mind, WDYT?

Reply:
Sure, we can test that in a potential future legalization pass.